acquia/blt
# Multisite setup enhancements and bugfixes

Fixes #2907, #2996, #2814, #2930    

Changes proposed:
- [x]  Validate mysql connection with output of drush status rather than drush sqlq for `blt internal:drupal:install` and `blt drupal:install`
- [x]  Create database with drush credentials if it does not exist
- [x]  Grants needed mysql user / db permissions for all multisite database names for all hosts including localhost
- [x]  Remove drupal_* namespace restriction for mysql database name grants
- [x]  Create sites.php file if it does not exist
- [x]  Generate sites.php sites array mapping project multisite local uris to site directories
- [x]  Remove default local.* config/settings files from new multisite directory to regenerate db and drush config from input
- [x]  Fix multisite local drush alias config
- [x]  Populate multsite database credentials from blt and user input
- [x]  Simplify use of $acsf_site_name in multisite setup
- [x]  Exclude default site files from multisite directory

Testing steps

Generate multisite site files, vm config, and settings.php config

 1. run `blt multisite`. Validate that the task: 
 - creates new multisite in docroot/sites/[sitename]
 - generates vm host in box/config.yml
 - generates vm db config in box.config.yml
 - generates local.settings.php and local.drush.yml with new multisite config
 - generates sites.php with new multisite mapping local site hostname to site directory:
 - Excludes files directory from default site
 - Creates drush alias in drush/sites/[sitename].site.yml
 - drush alias returns correct uri, site directory, and db credentials

2. Re-provision vm to add new multisite host and database
- Exit from vm and reprovision with `vagrant reload --provision`
- ssh to vm: `vagrant ssh`

3. Multisite install/setup
- Install drupal / set up blt: `blt setup --site=[sitename]`
- Validate that database is dropped and created automatically
- Validate that site is installed according to config in sites/[sitename].blt.yml / local.settings.php
- Once installed, both `drush @sitename.local st` and `drush --root=path/to/docroot --uri=hostname` should return the correct site uri and site path and the database in `sites/[sitename]/settings/local.settings.php` should bootstrap
